tasks/request_files/request_files.py

```diff
@@ -260,23 +300,26 @@ def inner_task(
                 FILE_REQUEST_TIME_KEY: time_stamp,
                 FILE_LAST_UPDATE_KEY: time_stamp,
             }
-            if object_exists(s3, glacier_bucket, file_key):
+            file_info = get_s3_object_information(s3, glacier_bucket, file_key)
+            if file_info is not None:
+                if file_info["StorageClass"] == "DEEP_ARCHIVE" and recovery_type == "Expedited":
+                    raise RestoreRequestError(f"File '{file_key}' from bucket '{glacier_bucket}' "
+                                              f"is in storage class '{file_info['StorageClass']}' "
+                                              f"which is incompatible with recovery type '{recovery_type}'")
                 LOGGER.info(
                     f"Added {file_key} to the list of files we'll attempt to recover."
                 )
             else:
-                message = f"{file_key} does not exist in {glacier_bucket} bucket"
+                message = f"'{file_key}' does not exist in '{glacier_bucket}' bucket"
                 LOGGER.error(message)
                 a_file[FILE_SUCCESS_KEY] = True
                 a_file[FILE_STATUS_ID_KEY] = shared_recovery.OrcaStatus.FAILED.value
                 a_file[FILE_ERROR_MESSAGE_KEY] = message
                 a_file[FILE_COMPLETION_TIME_KEY] = time_stamp
             files.append(a_file)
 
-        # Create a copy of the granule and add file information in the proper
-        # format
-        copied_granule = granule.copy()
-        copied_granule[GRANULE_RECOVER_FILES_KEY] = files
+        # Add file information in the proper format
+        granule[GRANULE_RECOVER_FILES_KEY] = files
```

Comment:
Any fallout from not using a copy of the object?

Reply:
Nope! The original granule, including the granules array, is not used aside from this one spot.